homeassistant/components/template/fan.py

```diff
@@ -344,6 +345,8 @@ async def async_update(self):
         # Validate state
         if state in _VALID_STATES:
             self._state = state
+        elif state == STATE_UNAVAILABLE:
+            self._available = False
         elif state == STATE_UNKNOWN:
```

Comment:
I'm reading the docs for the template fan and thinking we should not change the availability feature since there's already the availability template and this is the same for all template platforms.

https://www.home-assistant.io/integrations/fan.template/#availability_template

It would be safer to have the same behavior as for `STATE_UNKNOWN` for now. If we want to change this we should make an architecture issue and decide an approach for all template platforms.

Sorry for the back and forth on this. I should have read up better earlier.

```suggestion
        elif state in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
```